DQM/SiOuterTracker/plugins/OuterTrackerMonitorTTCluster.cc

```diff
@@ -96,24 +104,17 @@ OuterTrackerMonitorTTCluster::~OuterTrackerMonitorTTCluster() {
 //
 // member functions
 //
+void OuterTrackerMonitorTTCluster::dqmBeginRun(const edm::Run &iRun, const edm::EventSetup &iSetup) {
+  tkGeom_ = &(iSetup.getData(geomToken_));
+  tTopo_ = &(iSetup.getData(topoToken_));
+}
 
 // ------------ method called for each event  ------------
 void OuterTrackerMonitorTTCluster::analyze(const edm::Event &iEvent, const edm::EventSetup &iSetup) {
   /// Track Trigger Clusters
   edm::Handle<edmNew::DetSetVector<TTCluster<Ref_Phase2TrackerDigi_>>> Phase2TrackerDigiTTClusterHandle;
   iEvent.getByToken(tagTTClustersToken_, Phase2TrackerDigiTTClusterHandle);
 
-  /// Geometry
-  edm::ESHandle<TrackerTopology> tTopoHandle;
-  const TrackerTopology *tTopo;
-  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
-  tTopo = tTopoHandle.product();
-
-  edm::ESHandle<TrackerGeometry> tGeometryHandle;
-  const TrackerGeometry *theTrackerGeometry;
-  iSetup.get<TrackerDigiGeometryRecord>().get(tGeometryHandle);
-  theTrackerGeometry = tGeometryHandle.product();
```

Comment:
Just to note that the recommended pattern for getting ES products (in cases like this) is to get them in `analyze()` instead of `beginRun()`. That would e.g. lead to simpler code (no need for new member variables or to specify `edm::Transition::BeginRun`) and work automatically for IOVs that have LuminosityBlock boundaries.